processor/src/test/java/org/mapstruct/ap/test/collection/iterabletononiterable/IterableToNonIterableMappingTest.java

```diff
@@ -45,4 +47,34 @@ public void shouldReverseMapStringListToStringUsingCustomMapper() {
         assertThat( source.getNames() ).isEqualTo( Arrays.asList( "Alice", "Bob", "Jim" ) );
         assertThat( source.publicNames ).isEqualTo( Arrays.asList( "Alice", "Bob", "Jim" ) );
     }
+
+    @Test
+    @IssueKey("607")
+    @WithClasses({
+            FruitsMapper.class
```

Comment:
You can move this in the class annotated `WithClasses`. This might speed up the test a bit since it will do only one compilation per class instead of doing multiple in case the methods are ordered differently. Same comment for the other test methods.